entity-service-impl/src/main/java/org/hypertrace/entity/query/service/EntityQueryServiceImpl.java

```diff
@@ -650,4 +731,19 @@ private Converter<List<Expression>, Selection> getSelectionConverter() {
                 Converter<
                     EntityQueryRequest, org.hypertrace.core.documentstore.query.Query>>() {}));
   }
+
+  private boolean validateDeleteEntitiesRequest(
+      DeleteEntitiesRequest request, RequestContext requestContext) {
+    Optional<String> tenantId = requestContext.getTenantId();
+    if (tenantId.isEmpty()) {
+      LOG.error("{}. Invalid deleteEntities request: Tenant id is not provided", request);
```

Comment:
instead of logging an error and return a boolean, instead throw and let the caller decide what to do in case of an error (it should be returning that error to the caller and logging it at the top level).